sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-supplier-integration.scheduler.e2e-spec.ts

```diff
@@ -679,4 +687,166 @@ describe(describeProcessorRootTest(QueueNames.CASSupplierIntegration), () => {
       ]),
     ).toBe(true);
   });
+
+  it("Should insert a record in the casSuppliers table when getSupplierInfoFromCAS throws a known error.", async () => {
+    // Arrange
+    // Created a student with same address line 1 and postal code from the expected CAS mocked result.
+    // Postal code has a white space that is expected to be removed.
+    const student = await saveFakeStudent(db.dataSource);
+    const savedCASSupplier = await saveFakeCASSupplier(db, { student });
+
+    // Configure CAS mock to return a custom named Bad Request error.
+    casServiceMock.getSupplierInfoFromCAS = jest.fn().mockImplementation(() => {
+      throw new CustomNamedError("CAS Bad Request Errors", CAS_BAD_REQUEST, [
+        "0035: Site does not exist.",
+      ]);
+    });
+
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+
+    // Act
+    const result = await processor.processCASSupplierInformation(mockedJob.job);
+
+    // Assert
+    expect(result).toStrictEqual([
+      "Process finalized with success.",
+      "Pending suppliers to update found: 1.",
+      "Records updated: 0.",
+      "Attention, process finalized with success but some errors and/or warnings messages may require some attention.",
+      "Error(s): 0, Warning(s): 2, Info: 10",
+    ]);
+    // Assert DB was updated.
+    const updateCASSupplier = await db.casSupplier.findOne({
+      select: {
+        id: true,
+        isValid: true,
+        supplierStatus: true,
+        errors: true,
+      },
+      where: {
+        id: savedCASSupplier.id,
+      },
+    });
+    expect(updateCASSupplier).toEqual({
+      id: savedCASSupplier.id,
+      isValid: false,
+      supplierStatus: SupplierStatus.ManualIntervention,
+      errors: ["0035: Site does not exist."],
+    });
+  });
+
+  it("Should insert a record in the casSuppliers table when createSiteForExistingSupplier throws a known error.", async () => {
```

Comment:
This E2E will need to be adapted to as per https://github.com/bcgov/SIMS/pull/4017#discussion_r1870059940